src/com/lftechnology/java/training/sanish/application/model/domain/User.java

```diff
@@ -0,0 +1,157 @@
+
+import com.lftechnology.java.training.sanish.application.component.Constants;
+import com.lftechnology.java.training.sanish.application.model.Model;
+
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * User model
+ *
+ * @author Sanish Maharjan <sanishmaharjan@lftechnology.com>
+ */
+public class User implements Model<User> {
+    private static final Logger LOGGER = Logger.getLogger(User.class.getName());
+    private int userId;
+    private String userName;
+    private String password;
+    private String email;
+    private boolean isTerminated;
+    private String createdAt;
+    private String modifiedAt;
+
+    public void setUserId(int userId) {
+        this.userId = userId;
+    }
+
+    public void setUserName(String userName) {
+        this.userName = userName;
+    }
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    public void setIsTerminated(boolean isTerminated) {
+        this.isTerminated = isTerminated;
+    }
+
+    public void setEmail(String email) {
+        this.email = email;
+    }
+
+    public void setCreatedAt(String createdAt) {
+        this.createdAt = createdAt;
+    }
+
+    public void setModifiedAt(String modifiedAt) {
+        this.modifiedAt = modifiedAt;
+    }
+
+    public int getUserId() {
+        return userId;
+    }
+
+    public String getModifiedAt() {
+        return modifiedAt;
+    }
+
+    public String getCreatedAt() {
+        return createdAt;
+    }
+
+    public boolean isTerminated() {
+        return isTerminated;
+    }
+
+    public String getEmail() {
+        return email;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public String getUserName() {
+        return userName;
+    }
+
+    @Override public String getTable() {
+        return "users";
+    }
+
+    @Override public <V> V getAttribute(String key) {
+        try {
+            return (V) User.class.getDeclaredField(key).get(this);
+        } catch (NoSuchFieldException e) {
+            LOGGER.log(Level.WARNING, Constants.EXCEPTION_ERROR_MSG_LABEL + "{0}", new Object[] { e });
+        } catch (IllegalAccessException e) {
+            LOGGER.log(Level.WARNING, Constants.EXCEPTION_ERROR_MSG_LABEL + "{0}", new Object[] { e });
+        }
+        return null;
+    }
+
+    @Override public <V> void setAttribute(String key, V value) {
+        try {
+            User.class.getDeclaredField(key).set(this, value);
+        } catch (NoSuchFieldException e) {
+            LOGGER.log(Level.WARNING, Constants.EXCEPTION_ERROR_MSG_LABEL + "{0}", new Object[] { e });
+        } catch (IllegalAccessException e) {
+            LOGGER.log(Level.WARNING, Constants.EXCEPTION_ERROR_MSG_LABEL + "{0}", new Object[] { e });
+        }
+    }
+
+    @Override public boolean equals(User userModel) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either override Object.equals(Object), or totally rename the method to prevent any confusion. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1201)